lib/sensors/led_driver.cpp

```diff
@@ -0,0 +1,97 @@
+#include "led_driver.hpp"
+
+namespace hyped::sensors {
+
+std::optional<LedDriver> LedDriver::create(core::ILogger &logger,
+                                           std::shared_ptr<io::II2c> i2c,
+                                           const std::uint8_t device_address)
+{
+  if (device_address != kDefaultLedDriverAddress) {
+    logger.log(core::LogLevel::kFatal, "Invalid device address for LED driver");
+    return std::nullopt;
+  }
+
+  auto ledDriver = LedDriver(logger, i2c, device_address);
+  if (auto init_result = ledDriver.initialise(); !init_result) {
+    logger.log(core::LogLevel::kFatal, "Failed to initialise LED driver");
+    return std::nullopt;
+  }
+
+  logger.log(core::LogLevel::kDebug, "Successfully initialised LED driver");
+
+  return ledDriver;
+}
+
+LedDriver::LedDriver(core::ILogger &logger,
+                     std::shared_ptr<io::II2c> i2c,
+                     const std::uint8_t device_address)
+    : logger_(logger),
+      i2c_(i2c),
+      device_address_(device_address)
+{
+}
+
+LedDriver::~LedDriver()
+{
+}
+
+std::optional<core::Result> LedDriver::initialise()
```

Comment:
this doesn't need to be `optional`, just return `core::Result::kFailure`/`core::Result::kSuccess`

Reply:
same for all the others that currently return `std::optional<core::Result>`